langchain/src/chat_models/googlevertexai.ts

```diff
@@ -0,0 +1,300 @@
+import { BaseChatModel } from "./base.js";
+import {
+  AIChatMessage,
+  BaseChatMessage,
+  ChatGeneration,
+  ChatResult,
+  LLMResult,
+  MessageType,
+  SystemChatMessage,
+} from "../schema/index.js";
+import { BaseLanguageModelCallOptions } from "../base_language/index.js";
+import { GoogleVertexAIConnection } from "../util/googlevertexai-connection.js";
+import {
+  GoogleVertexAIBaseLLMInput,
+  GoogleVertexAIBasePrediction,
+  GoogleVertexAILLMResponse,
+  GoogleVertexAIModelParams,
+} from "../types/googlevertexai-types.js";
+
+/**
+ * Represents a single "example" exchange that can be provided to
+ * help illustrate what a model response should look like.
+ */
+export interface ChatExample {
+  input: BaseChatMessage;
+  output: BaseChatMessage;
+}
+
+interface GoogleVertexAIChatExample {
+  input: GoogleVertexAIChatMessage;
+  output: GoogleVertexAIChatMessage;
+}
+
+export type AuthorType = "user" | "bot";
+
+export interface GoogleVertexAIChatMessage {
+  author: AuthorType;
+  content: string;
+  name?: string;
+}
+
+export interface GoogleVertexAIChatInstance {
+  context?: string;
+  examples?: GoogleVertexAIChatExample[];
+  messages: GoogleVertexAIChatMessage[];
+}
+
+export interface GoogleVertexAIChatPrediction
+  extends GoogleVertexAIBasePrediction {
+  candidates: GoogleVertexAIChatMessage[];
+}
+
+export interface GoogleVertexAIChatInput extends GoogleVertexAIBaseLLMInput {
+  /** Instructions how the model should respond */
+  context?: string;
+
+  /** Help the model understand what an appropriate response is */
+  examples?: ChatExample[];
+
+  /**
+   * A map of OpenAI role names and their corresponding Vertex AI
+   * author name.
+   */
+  roleAlias?: RoleAlias;
+}
+
+export type RoleAlias = Record<MessageType, AuthorType | undefined>;
+
+export interface GoogleVertexAIChatCallOptions
+  extends BaseLanguageModelCallOptions {
+  context?: string;
+
+  examples?: ChatExample[];
+}
+/**
+ * Enables calls to the Google Cloud's Vertex AI API to access
+ * Large Language Models in a chat-like fashion.
+ *
+ * To use, you will need to have one of the following authentication
+ * methods in place:
+ * - You are logged into an account permitted to the Google Cloud project
+ *   using Vertex AI.
+ * - You are running this on a machine using a service account permitted to
+ *   the Google Cloud project using Vertex AI.
+ * - The `GOOGLE_APPLICATION_CREDENTIALS` environment variable is set to the
+ *   path of a credentials file for a service account permitted to the
+ *   Google Cloud project using Vertex AI.
+ */
+export class ChatGoogleVertexAI
+  extends BaseChatModel
+  implements GoogleVertexAIChatInput
+{
+  declare CallOptions: GoogleVertexAIChatCallOptions;
+
+  model = "chat-bison";
+
+  temperature = 0.2;
+
+  maxOutputTokens = 256;
+
+  topP = 0.8;
+
+  topK = 40;
+
+  context: string;
+
+  examples: ChatExample[];
+
+  roleAlias: RoleAlias = {
```

Comment:
Having a look here: https://cloud.google.com/vertex-ai/docs/generative-ai/chat/chat-prompts

It looks like `SystemChatMessages` could be mapped to a vertex message that has an `author` set to `context`? We can avoid the special case below if so.